shield/internal/lexer/lexer.go

```diff
@@ -141,17 +147,17 @@ func (l *Lexer) readNumber() string {
 	return l.input[position:l.position]
 }
 
-func (l *Lexer) readString() string {
+func (l *Lexer) readString() (string, bool) {
```

Comment:
I'd change the return type to `token.Token`
On the other hand it will differ from the other similar methods like `readNumber`

Reply:
done as you said, this file will need a refactor anyway at some point :)